pages/Messages/Signatures.md

```diff
@@ -0,0 +1,44 @@
+---
+name: Signatures
+route: /Messages/Signatures
+menu: Messages
+---
+
+# Message Signatures
+
+All DSNP messages provided for batching should include a [secp256k1](https://google.com/search?hl=en&q=secp256k1) signature for the purposes of verifying the authenticity of the message.
+This signature should be generated from the [keccak-256](https://en.wikipedia.org/wiki/SHA-3) hash of the message contents and the publishing user's private key.
+Optionally, anonymous messages may be provided with a zero hash in place of the signature, however behavior for these messages is undefined, and they may be treated as invalid data by archivists or disregarded as spam by indexers or end clients.
+
+## Specification Status
+
+| Version | Status |
+---------- | ---------
+| 0.1     | Tentative |
+
+## Purpose
+1. Describe the process of signing DSNP messages for inclusion in batches
+1. Describe the process of verifying DSNP messages from batches
+
+## Assumptions
+* All assumptions from [DSNP Messages](/Messages/Overview)
+
+## Signing Messages
+
+The process for generating signatures for DSNP messages consists of three distinct steps:
+
+1. Serialize the message
+1. Hash the serialized string
+1. Sign the hash
+
+Before serializing, keys in the DSNP message should be sorted alphabetically using [default JavaScript string comparison](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) order.
```

Comment:
I thought we were getting rid of this requirement and accepting that there may be different hashes for the same message.

Reply:
We removed the requirement for the hash of the activity pub content, since we don't control the keys in an activity pub object and sorting them becomes complex, but we still need this requirement for the hash of the DSNP message to guarantee we always get a consistent result.

Plus, it shouldn't be a problem for the DSNP object, since we control the all the keys in it and can guarantee they're always ANSI characters, which we can sort.